providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithClientConfigurer.java

```diff
@@ -0,0 +1,169 @@
+package dev.openfeature.contrib.providers.flagsmith;
+
+import com.flagsmith.FlagsmithClient;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.config.Retry;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidCacheOptionsExceptions;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.InvalidOptionsExceptions;
+
+/**
+ * FlagsmithClientConfigurer helps set up and validate the options for the FlagsmithClient
+ * used by the FlagsmithProvider class.
+ */
+public class FlagsmithClientConfigurer {
+
+    /**
+     * Check the options that have been provided to see if there are any issues.
+     * Exceptions will be thrown if there are issues found with the options.
+     *
+     * @param options the options used to create the provider
+     */
+    private static void validateOptions(FlagsmithProviderOptions options) {
```

Comment:
Being picky, I prefer to see public methods above all private methods.

Reply:
Moved below the public methods